components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -405,17 +430,25 @@ func setPrefixEnvVar(instance *v1beta1.Notebook, container *corev1.Container) {
 	})
 }
 
-func generateStatefulSet(instance *v1beta1.Notebook) *appsv1.StatefulSet {
+func generateStatefulSet(instance *v1beta1.Notebook, isGenerateName bool) *appsv1.StatefulSet {
 	replicas := int32(1)
 	if metav1.HasAnnotation(instance.ObjectMeta, "kubeflow-resource-stopped") {
 		replicas = 0
 	}
 
+	ssObjectMeta := metav1.ObjectMeta{
+		Name:      instance.Name,
+		Namespace: instance.Namespace,
+	}
+	if isGenerateName {
+		ssObjectMeta = metav1.ObjectMeta{
+			GenerateName: "nb-",
```

Comment:
How will this look like in the real world? isn't it too generic? I mean - would it make sense to have at least some portion of the actual original `instance.Name` here so it's at least somehow traceable to the original user via `oc get ...` command?

Reply:
> How will this look like in the real world? isn't it too generic?

As this change would only happen if `isGenerateName` is set, which depends on notebook name to be larger than 52 char. In ODH scenario, it is only possible via jupyter tile , as DS project has slices notebook name above 30 char. 
In jupyter tile, user dont get to set the notebook name on the UI, so on console it would show up as nb-<hash>
user can query them via cli using:
`oc get notebook --all-namespaces -l opendatahub.io/user={username}`
`
or 
`
oc get statefulset --all-namespaces -l statefulset={notebook.name}
`

we could have some part of instance.name, i wasn't sure, what slice should it be prefix or suffix end of instance.name.

Reply:
Absolutely, i do feel inline with you. 
With this case being exclusive to jupyter tiles, keeping the name sequence could be a way. Perhaps we can do this in next iterations.